subprojects/files/src/main/java/org/gradle/api/internal/file/archive/impl/FileZipInput.java

```diff
@@ -52,16 +52,13 @@ public static ZipInput create(File file) {
      * {@link ZipFile} is more efficient, but causes memory leaks on older Java versions, so we only use it on more recent ones.
      */
     private static boolean isZipFileSafeToUse() {
-        return getJavaMajorVersion() >= 11;
-    }
-
-    private static int getJavaMajorVersion() {
-        String versionString = System.getProperty("java.version");
+        String versionString = System.getProperty("java.specification.version");
+        // Java versions 8 and older had 1.8 versioning scheme, later ones have single number 9, 10, ...
         String[] versionParts = versionString.split("\\.");
         if (versionParts.length < 1) {
             throw new IllegalArgumentException("Could not determine java version from '" + versionString + "'.");
         }
-        return Integer.parseInt(versionParts[0]);
+        return Integer.parseInt(versionParts[0]) >= 11;
```

Comment:
Shouldn't we just reuse `JavaVersion`? I understand it wasn't done because `baseServices` isn't a dependency of `files`, but it looks like the right thing to do to me (either move JavaVersion to another utility project, or depend on base services). Because `JavaVersion` doesn't have this bug, and otherwise we're going to have to fix this again some time in the future...

Reply:
I thought there was some reason why it can't depend on `baseServices` as it was recently made not to use `JavaVersion` https://github.com/gradle/gradle/pull/13283/files#diff-5621357e146b9a03893a59e566f04d66

It is probably fine to add this dependency - let's see if it builds.

Reply:
This code is shared with Maven caching, and Maven caching does not depend on `:base-services` (nor should it). We could create some kind of `:base-base-services` or `:utils` project to share between the two products, though, and move `JavaVersion` there, although `JavaVersion` itself has some baggage, too, that we might not want to share.

I didn't realize the switch from `JavaVersion` to this code was recent, though I should have. I think I'd have recommended sharing `JavaVersion` instead.

Reply:
Yeah, changing this was a part of moving this class to the `files` project to share it with the Maven cache extension.  Sharing `JavaVersion` meant either moving that class to `files` (which didn't seem to make sense) or creating a new project (and dependency jar) with basically only `JavaVersion` in it (which seemed like massive overkill).  In the end, this was about 10 lines of code to parse a string, so it seemed reasonable to just embed a simplified version in this class.  I'd definitely prefer to use the logic in `JavaVersion`, so if we can find a reasonable way to share that code, I think that would be better.